IOPool/Input/test/testNoParentDictionary.sh

```diff
@@ -0,0 +1,45 @@
+#!/bin/bash -ex
+
+function die { echo $1: status $2 ;  exit $2; }
+
+SCRAM_TEST_NAME=TestIOPoolInputNoParentDictionary
+rm -rf $SCRAM_TEST_NAME
+mkdir $SCRAM_TEST_NAME
+cd $SCRAM_TEST_NAME
+
+cmsRun ${LOCALTOP}/src/IOPool/Input/test/PoolNoParentDictionaryTestStep1_cfg.py || die 'Failed cmsRun PoolNoParentDictionaryTestStep1_cfg.py' $?
+
+# Create a new CMSSW dev area
+OLD_CMSSW_BASE=${CMSSW_BASE}
+scram -a $SCRAM_ARCH project $CMSSW_VERSION
+pushd $CMSSW_VERSION/src
+eval `scram run -sh`
+
+# Copy DataFormats/TestObjects from local/release
+# Plus minimal set of packages to run the test for the first time
+# TODO: this test doesn't really work on PRs that have code outside of
+# these packages that would break the functionality
```

Comment:
Any ideas to improve this test are welcome. I tried to to symlink all files (except those matching to `DataFormatsTestObjects`) from `${OLD_CMSSW_BASE}/lib/${SCRAM_ARCH}` into `${CMSSW_BASE}/lib/${SCRAM_ARCH}` but that alone was not sufficient (the step2 job below failed as the code changes were not propagated).

_Maybe_ this test could be sufficient until @smuzaffar gets back.

Reply:
How about just manipulating what the PluginManager reads instead of having to do another compilation?
1. Put the dictionary generation of TransientIntParentT<1> and TransientIntParentT<2> into two different classes_def files so they are in different plugins
2. Write using TransientIntParentT<1> as you did
3. manipulate LD_LIBRARY_PATH to add a new directory to the beginning which has a .edmplugincache which lies about where to get TransientIntParentT<1> (points to a file that doesn't contain that dictionary)

Reply:
I'm not sure I understood properly. The dictionaries end up in libraries, not plugins (right?). I can't quickly think of a way to play tricks with libraries and `LD_LIBRARY_PATH` without a second compilation, but maybe adjusting `LD_LIBRARY_PATH` with the second compilation would at least help to overcome the main deficiency of the current test.

Reply:
Arg, I was thinking about ancient days where plugin manager provided the dictionaries. Still, the same idea might work if one manipulated a .rootmap file and LD_LIBRARY_PATH.

Reply:
I finally figured out one way to improve the test such that it should work now in any PR/IB, fails in the IB without this PR, and does not require `scram b`.